pyDataverse/auth.py

```diff
@@ -0,0 +1,103 @@
+"""This module contains authentication handlers compatible with :class:`httpx.Auth`"""
+
+from typing import Generator
+
+from httpx import Auth, Request, Response
+
+from pyDataverse.exceptions import ApiAuthorizationError
+
+
+class ApiTokenAuth(Auth):
+    """An authentication handler to add an API token as the X-Dataverse-key
+    header.
+
+    For more information on how to retrieve an API token and how it is used,
+    please refer to https://guides.dataverse.org/en/latest/api/auth.html.
+    """
+
+    def __init__(self, api_token: str):
+        """Initializes the auth handler with an API token.
+
+        Parameters
+        ----------
+        api_token : str
+            The API token retrieved from your dataverse instance user profile.
+
+        Examples
+        --------
+
+            >>> import os
+            >>> from pyDataverse.api import DataAccessApi
+            >>> base_url = 'https://demo.dataverse.org'
+            >>> api_token_auth = ApiTokenAuth(os.getenv('API_TOKEN'))
+            >>> api = DataAccessApi(base_url, api_token_auth)
+
+        """
+        if not isinstance(api_token, str):
+            raise ApiAuthorizationError("Api token passed is not a string.")
```

Comment:
```suggestion
            raise ApiAuthorizationError("API token passed is not a string.")
```

Reply:
Hm, I kept the original error message in case someone relied on it, but I think this is a sensible change.